php-frontend/src/main/java/org/sonar/plugins/php/api/cfg/ControlFlowGraph.java

```diff
@@ -117,8 +121,8 @@ public static ControlFlowGraph build(Tree tree, CheckContext context) {
           throw new IllegalStateException("Unexpected tree kind " + tree.getKind());
       }
     } catch (Exception e) {
-      LOG.warn("Failed to build control flow graph for file [{}] at line {}", context.getPhpFile().toString(), ((PHPTree) tree).getLine());
-      LOG.warn(e.getMessage(), e);
+      LOG.warn("Failed to build control flow graph for file [{}] at line {} caused by : {}", context.getPhpFile().toString(), ((PHPTree) tree).getLine(), e.getCause());
```

Comment:
`e.getCause()` returns a stacktrace of a root exception, probably that's not what we want (as it's still huge and too low level to give any information). Probably `e.getMessage()` will do a better job.

Reply:
I believe that if we make that choice then we'll just mention that for more detail you want to activate debug logs.

Reply:
I like that option